### ISSUE TITLE
Dealing with multiple async values (streams / async iterators)

### ISSUE THREAD

**Issue by cryptogohan** [nullpub/fun#10]
What do you recommend?

Any plans to support?

Been using this library for quite a few hours now. Loving it ❤️  !

**Comment by baetheus** [comment]
I've done a lot of work using [rxjs](https://rxjs.dev/), [mostjs](https://github.com/mostjs/core), and [callbags](https://github.com/callbag/callbag) in the browser and nodejs environments so that's my js experience with async streams. 

Building a stream library is an interesting problem. Ideally I'll find the time to implement something in fun but I haven't quite settled on the feature set yet. I took a [shot at callbags](https://gist.github.com/baetheus/5a8012373e863cb7b157bef8505fabd5) about 6 months ago and it seemed possible, but I'd probably want to do a little more research before settling.

That said, I don't yet have a recommendation for a stream library on deno, let alone a "functional" one. I only have vague plans and interest in taking a shot at my own design. We are lucky in that the deno runtime is different enough that there is space to move away from existing libraries (like I did in fun) and try something new. That said, existing streaming libs are pretty battle tested and performant (I'm thinking mostjs here) so it would be smart to learn from them where possible. At the end of the day I think anything I implement will need to have mechanisms for handling backpressure, which I haven't really taken the time to understand completely. So.. don't hold your breath :D
